plugins/device/device.proto

```diff
@@ -0,0 +1,108 @@
+syntax = "proto3";
+package hashicorp.nomad.plugins.device;
+
+import "google/protobuf/empty.proto";
+
+// DevicePlugin is the API exposed by device plugins
+service DevicePlugin {
+  // FingerprintDevices allows the device plugin to return a set of   
+  // detected devices and provide a mechanism to update the state of
+  // the device.
+  rpc FingerprintDevices(google.protobuf.Empty) returns (stream DetectedDevices) {}
+
+  // Allocate is called by the client before starting an allocation
+  // that requires access to the plugin’s devices. The plugin can use
+  // this to run any setup steps and provides the mounting details to 
+  // the Nomad client
+  rpc Allocate(AllocateRequest) returns (AllocateResponse) {}
+}
+
+// DetectedDevices is the set of devices that the device plugin has 
+// detected and is exposing
+message DetectedDevices {
+  // device_name is the name of the device. It should be scoped by 
+  // vendor  (“nvidia/gpu”, “intel/gpu”, “amd/gpu”, …)
+  string device_name = 1;
```

Comment:
Is the plan to have the device be opaque to the scheduler? Meaning for example a job requests a `foo` device and the scheduler will need to go and see if a `foo` device has been fingerprinted somewhere? 

I ask because I wonder if it would be better to have a string vendor name and an enum of device types, ex `gpu`, `rdma`, etc.

Reply:
That is the plan. I think the enum just is a maintenance burden since it is opaque to Nomad, so may as well be a string. But I like the idea of separating vendor and device type. Will make them two fields